sklearn/utils/__init__.py

```diff
@@ -58,9 +58,12 @@
 
 
 class Bunch(dict):
-    """Container object for datasets
+    """Container object exposing keys as attributes
 
-    Dictionary-like object that exposes its keys as attributes.
+    ``Bunch`` is a dictionary-like object that exposes its keys as attributes.
+    It is sometimes used as an output and sometimes to store certain
+    information in attributes of certain objects. This object should not be
+    used as an input parameter.
```

Comment:
> sometimes to store certain    information in attributes of certain objects

What cases is this referring to?

> This object should not be    used as an input parameter.

Isn't this one of these cases where we're just giving bad ideas? I.e. no one would have thought of that in the first place? Just wondering

Reply:
> Isn't this one of these cases where we're just giving bad ideas? I.e. no one would have thought of that in the first place? Just wondering

It's in the FAQ, doesn't hurt to tell people not to do that.

> What cases is this referring to?

`named_estimator_` for instance.